beacon-chain/sync/initial-sync/service.go

```diff
@@ -239,3 +201,38 @@ func (s *Service) waitForMinimumPeers() {
 		time.Sleep(handshakePollingInterval)
 	}
 }
+
+// waitForStateInitialization makes sure that beacon node is ready to be accessed: it is either
+// already properly configured or system waits up until state initialized event is triggered.
+func (s *Service) waitForStateInitialization() (time.Time, error) {
+	headState, err := s.chain.HeadState(s.ctx)
+	if err == nil && headState != nil {
```

Comment:
you need to handle the case where `err != nil`  also

Reply:
How should I handle such a case? Originally, if error wasn't nil, the case was ignored and wait for event starts. This refactored implementation mimics exactly this behaviour. 

Do you want to halt on non-nil error and return from the function with that error (in which case `Start()` will also be halted)?

Reply:
Error is explicitly returned now.